google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardTableDefinition.java

```diff
@@ -272,7 +272,31 @@ static StandardTableDefinition fromPb(Table tablePb) {
       builder.setStreamingBuffer(StreamingBuffer.fromPb(tablePb.getStreamingBuffer()));
     }
     if (tablePb.getTimePartitioning() != null) {
-      builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+      try {
+        builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(
+            "Illegal Argument - Got unexpected time partitioning "
+                + tablePb.getTimePartitioning().getType()
```

Comment:
Actually, can we output the whole `timePartitioning` pretty-printed to a string?

Reply:
That'll give us a little more context

Reply:
changing to using toString()